webapp/channels/src/utils/i18n.tsx

```diff
@@ -2,7 +2,25 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import {FormattedMessage, type IntlShape, type MessageDescriptor} from 'react-intl';
+import {FormattedMessage, createIntl, createIntlCache, type IntlShape, type MessageDescriptor} from 'react-intl';
+
+import {getCurrentLocale, getTranslations} from 'selectors/i18n';
+import store from 'stores/redux_store';
+
+const cache = createIntlCache();
+
+// getIntl returns a memoized instance of IntlShape for the current locale.
+// Prefer `useIntl` and `FormattedMessage` and only use this selectively when
+// outside of React components.
+export function getIntl(): IntlShape {
```

Comment:
Is the only place where this is used in the `SwitchChannelProvider`? If so, I think we can continue to avoid adding this. I really want to avoid adding this since it's going to have one of the same issues that `localizeMessage` has where it won't properly update the rendered text when the locale changes.

If we do still need it though, I don't think this is actually memoizing the returned value itself though. If you were to turn it into a selector using `createSelector` and then have callers pass in the global store when they need it, it'll ensure that the object is memoized. That also has the nice side effect of not importing the global store into this file since I feel like that might affect code splitting slightly

Reply:
In this first PR, it is indeed the only place where it's used, but I needed to rely on it in other files yet to be submitted.

> I really want to avoid adding this since it's going to have one of the same issues that localizeMessage has where it won't properly update the rendered text when the locale changes.

I'm expecting we should only use this outside a React component, and my changes will dramatically reduce the scenarios today where we never needed `Utils.localizeMessage`. The "trick" with doing it this way is that `formatJS` will pick up on the `intl.formatMessage` even though it's not the "real" `intl`. Is it ok to solve just the `formatJS` extraction part, and defer the refactoring required to detect a locale change outside of a React component?

> If we do still need it though, I don't think this is actually memoizing the returned value itself though.

You're right -- I'll still keep the cache as-is, but drop the memoization part as the claim is too strong.

Reply:
> I'm expecting we should only use this outside a React component, and my changes will dramatically reduce the scenarios today where we never needed `Utils.localizeMessage`.

That's fair. I think we should still aim to get rid of this in the future to solve the "translating outside of React context" problem, but this does get us addressing the "we can't use FormatJS tooling" problem.